### PR TITLE
SAK-30727 avoid NPE and ensure redirect to /describe

### PR DESCRIPTION
**Pull request by steveswinsburg**

### CODE REVIEW

File: entitybroker/utils/src/java/org/sakaiproject/entitybroker/util/servlet/DirectServlet.java [hunk 1]
```diff
@@ -147,12 +147,13 @@ protected void handleRequest(HttpServletRequest req, HttpServletResponse res)
...
         // catch the login helper posts
         String option = req.getPathInfo();
-        String[] parts = option.split("/");
-        if ((parts.length == 2) && ((parts[1].equals("login")))) {
-            handleUserLogin(req, res, null);
-        } else {
-            dispatch(req, res);
-        }
+        if (option != null) {
+			final String[] parts = option.split("/");
```
Comment (buckett) [inline entitybroker/utils/src/java/org/sakaiproject/entitybroker/util/servlet/DirectServlet.java]: Indentation.

**Comment by steveswinsburg** [comment]
spaces v tabs.

File: entitybroker/utils/src/java/org/sakaiproject/entitybroker/util/servlet/DirectServlet.java [hunk 2]
```diff
@@ -147,12 +147,13 @@ protected void handleRequest(HttpServletRequest req, HttpServletResponse res)
...
-            handleUserLogin(req, res, null);
-        } else {
-            dispatch(req, res);
+        if (option != null) {
+            final String[] parts = option.split("/");
+            if ((parts.length == 2) && ((parts[1].equals("login")))) {
+                handleUserLogin(req, res, null);
+            }
         }
+        dispatch(req, res);
```
Comment (buckett) [inline entitybroker/utils/src/java/org/sakaiproject/entitybroker/util/servlet/DirectServlet.java]: Won't it be the case that when the login tool has rendered the page you don't want to dispatch it on?

So the `dispatch(req, res)` shouldn't be called when logging in I think.

**Comment by jonespm** [comment]
I believe this was resolved by #2291 and just should be closed.

**Pull request closed without merging**